garage/np/algos/cem.py

```diff
@@ -81,6 +87,8 @@ def sample_params(self, epoch):
             self.n_params) * sample_std + self.cur_mean
 
     def train_once(self, itr, paths):
+        paths = self.process_samples(itr, paths)
```

Comment:
Can we move this to the runner?

Reply:
hmm interesting design question. @naeioi what do you think?

Reply:
I prefer to give an algorithm more control over the batch data. That is, let the algorithm decides when and how to do preprocessing, or even not doing it at all.

Reply:
Hmm I think the algorithms are defining how to optimize the models for the system while the runner are defining the flow of everything. (e.g. sampling data -> preprocessing them -> feed them to the algorithms for optimization).

If the workflow of a training procedure is different, it sounds like a different runner for me.

Reply:
Moving the pre-processing to the runner makes more sense to me: it's better that `train_once()` do only training.

Reply:
This can always be refactored easily, so for now let's just go with Keren's judgement and see how it pans out.